alkira/resource_alkira_peering_gateway_cxp.go

```diff
@@ -0,0 +1,194 @@
+				Description: "The region of the specified cloud provider on " +
+					"which the resource should be created. E.g. if " +
+					"`cloud_provider` is `AZURE`, the region could be like " +
+					"`eastus`.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"segment_id": {
+				Description: "The ID of the segment that is associated with " +
+					"the resource.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"state": {
+				Description: "The state of the resource.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceAlkiraPeeringGatewayCxpCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	// INIT
+	api := alkira.NewPeeringGatewayCxp(m.(*alkira.AlkiraClient))
+
+	request, err := generateAlkiraCxpPeeringGatewayRequest(d, m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// CREATE
+	response, _, err, _ := api.Create(request)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(string(response.Id))
+
+	// WAIT FOR STATE
+	state := response.State
+
+	for state != "ACTIVE" {
+		resource, _, err := api.GetById(d.Id())
+		if err != nil {
+			return diag.Diagnostics{{
+				Severity: diag.Warning,
+				Summary:  "FAILED TO GET RESOURCE",
+				Detail:   fmt.Sprintf("%s", err),
+			}}
+		}
+
+		state = resource.State
+		time.Sleep(5 * time.Second)
+	}
+
+	return resourceAlkiraPeeringGatewayCxpRead(ctx, d, m)
+}
+
+func resourceAlkiraPeeringGatewayCxpRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	// INIT
+	api := alkira.NewPeeringGatewayCxp(m.(*alkira.AlkiraClient))
+
+	resource, _, err := api.GetById(d.Id())
+	if err != nil {
+		return diag.Diagnostics{{
+			Severity: diag.Warning,
+			Summary:  "FAILED TO GET RESOURCE",
+			Detail:   fmt.Sprintf("%s", err),
+		}}
+	}
+	segmentId, err := getSegmentIdByName(resource.Segment, m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.Set("name", resource.Name)
+	d.Set("description", resource.Description)
+	d.Set("cxp", resource.Cxp)
+	d.Set("cloud_provider", resource.CloudProvider)
+	d.Set("cloud_region", resource.CloudRegion)
+	d.Set("segment_id", segmentId)
+	d.Set("state", resource.State)
+
+	return nil
+}
+
+func resourceAlkiraPeeringGatewayCxpUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	// checking if only description field is changed
+	if d.HasChanges("name", "cloud_provider", "cloud_region", "segment_id") {
+		// return error if any field except description is changed.
+		return diag.Diagnostics{{
+			Severity: diag.Error,
+			Summary:  "Invalid Update!",
+			Detail:   "Only the description field can be updated.",
+		}}
+	}
+
+	// INIT
+	api := alkira.NewPeeringGatewayCxp(m.(*alkira.AlkiraClient))
+
+	request, err := generateAlkiraCxpPeeringGatewayRequest(d, m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// UPDATE
+	_, err, _ = api.Update(d.Id(), request)
+
```

Comment:
We need to check this `err` before return or just simplify this update. We don't need to support update at all. Just disable update return directly.